### PR TITLE
Add Azure support.

### PR DESCRIPTION
**Pull request by AbelHu**

### CODE REVIEW

**Comment by allomov** [comment]
Some time ago I came across the same issue. I was surprised that `GetServerName` in `FileMetadataSource` returned empty string https://github.com/cloudfoundry/bosh-agent/issues/16. It's nice that you fixed it.

File: infrastructure/file_metadata_service.go [hunk 1]
```diff
@@ -46,6 +45,10 @@ func (ms fileMetadataService) GetPublicKey() (string, error) {
 }
 
 func (ms fileMetadataService) GetInstanceID() (string, error) {
+	if ms.metaDataFilePath == "" {
+		return "", nil
+	}
+
```
Comment (allomov) [inline infrastructure/file_metadata_service.go]: I am sure that this condition is redundant. If file path is not set, or set to the wrong path, `ReadFile ` is going to return the error ((source)[https://github.com/cloudfoundry/bosh-agent/blob/master/infrastructure/file_metadata_service.go#L51]) and bosh-agent will fail. Failing in this case is expected behaviour.

File: platform/disk/sfdisk_partitioner.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func (p sfdiskPartitioner) Partition(devicePath string, partitions []Partition)
 
 	_, _, _, err := p.cmdRunner.RunCommandWithInput(sfdiskInput, "sfdisk", "-uM", devicePath)
 	if err != nil {
+		p.cmdRunner.RunCommandWithInput("d\n2\nd\nw\n", "fdisk", devicePath)
```
Comment (allomov) [inline platform/disk/sfdisk_partitioner.go]: Is this code related to the problem this PR solves?

Comment (AbelHu) [inline platform/disk/sfdisk_partitioner.go]: This is used to fix an issue we hit on Azure. I will separate them to two PRs. Thanks.

File: infrastructure/file_metadata_service.go (hunk 1 above)
Comment (AbelHu) [inline infrastructure/file_metadata_service.go]: If  the failing is expected behavior. I can remove it.

Comment (allomov) [inline infrastructure/file_metadata_service.go]: `GetInstanceID` returns error if something is wrong, the idea is that someone who calls this function is responsible for processing this case. In the same time, missing `metaDataFilePath` is an error and it will be processed by `ReadFile`, so I guess the condition should be removed.

File: platform/disk/sfdisk_partitioner.go (hunk 2 above)
Comment (AbelHu) [inline platform/disk/sfdisk_partitioner.go]: @allomov I separate them into two commits but they are still included in one PR. Is it OK?

Comment (allomov) [inline platform/disk/sfdisk_partitioner.go]: @AbelHu I think it would be better if you'll create another PR to add this line. Lets wait for comments from @cppforlife or @maximilien.

**Comment by mariash** [comment]
PR was merged here (without disk reformatting for now): https://github.com/cloudfoundry/bosh-agent/commit/7d0e933b71736c56d238c5a9969ee969b1258337
